BugFix: Impulsarse contra tu agarre

AdriAlvarez00 opened the pull request:
Se comprueba el angulo para ver que te impulsas desde, y no contra.

Adrian-de-Lucas-Gomez committed:
Desarmar jugadores

Se ha creado una constante "DISARM_IMPACT" la cual marca el limite para poder ser golpeado por un arma y que no te pase nada. Si la velocidad que lleva lo sobrepasa te quedarás desarmado en el caso de que tuvieras un arma.
Por ahora solo te desarman las Weapon puras como la pelota saltarina por el problema de Weapon y sus clases hijas.

Adrian-de-Lucas-Gomez committed:
Merge branch 'master' into 172599851-SoltarArmasAlImpactar

Adrian-de-Lucas-Gomez committed:
Desarmar por colision de objetos lanzables

Para evitar problemas con los diferentes tipos de armas se ha optado por diferenciar los obketos si tienen el componente ThrownByPlayer ya que esto nos permite saber de forma implicita que se lanza y como sabemos los objetos que lanzamos (Los cuales todos son armas) y por lo tanto los que te pueden desarmar.

Se ha puesto un margen para la velocidad del impacto para evitar desarmarte al tocar con un arma que estaba estática.

Adrian-de-Lucas-Gomez committed:
Ajustes en sprites para coherencia

Simplemente para adaptarlos a los acpectos que tienen las armas cuando están flotando

Adrian-de-Lucas-Gomez committed:
Merge branch 'master' into 172599851-SoltarArmasAlImpactar

Adrian-de-Lucas-Gomez committed:
Merge branch 'master' into 172599851-SoltarArmasAlImpactar

Adrian-de-Lucas-Gomez committed:
Cambios al router

Solo un sprite mas claro para que se vea mejor

Adrian-de-Lucas-Gomez committed:
Eliminado en hueco que se dejaba al agarrase

Ya se queda bien pegadito a la pared. Lo he probado en la caldera que es la situación mas compleja.

Adrian-de-Lucas-Gomez committed:
Mejoras al platano

Si han refinado los sprites y se han hecho todos sobre el mismo diseño de Hands.

Tambien he quitado una cosa innecesaria que puse en el agarre

Adrian-de-Lucas-Gomez committed:
Merge branch 'master' into 172599851-SoltarArmasAlImpactar

Adrian-de-Lucas-Gomez committed:
Prueba para evitar clipping

Es una solucion y ayuda a la hora de hacer impulsos desde la pared pues de aleja un poco de ella.

Adrian-de-Lucas-Gomez committed:
Creo que ahora si (Los agarres)

Esto es un sufrimiento pero creo que ya estando en una sola pared no se puede clipear al otro lado. No he cambiado el grosor de las paredes pero como sea necesario se hará.

Adrian-de-Lucas-Gomez committed:
Merge branch 'master' into 172599851-SoltarArmasAlImpactar

Adrian-de-Lucas-Gomez committed:
Por Jesus bendito ya va bien

He dejado lo de que al impulsarte te separe un poco de la pared, ya no es necesario pero creo que beneficia a la movilidad evitando que roces con la pared.

Adrian-de-Lucas-Gomez committed:
Eliminada comprobacion con normales

Era demasiado restrictivo ya que hay paredes en diagonal

Adrian-de-Lucas-Gomez committed:
Merge branch 'master' into 172599851-SoltarArmasAlImpactar

Adrian-de-Lucas-Gomez committed:
Merge de las manitas

Adrian-de-Lucas-Gomez committed:
Ahora si que si

Creo que ya por fin he dado con la forma idonea para hacer los acercamientos a la pared. Basicamente usando las normales del plano pero en sentido contrario.

Adrian-de-Lucas-Gomez committed:
Mejoras del agarre

Se ha establecido un parametro de tolerancia de ángulos para evitar acercar demasiado al jugador si tiene un angulo cercano al paralelo a la pared

Adrian-de-Lucas-Gomez committed:
Merge branch 'master' into 172599851-SoltarArmasAlImpactar

Adrian-de-Lucas-Gomez committed:
Variables con nombres buenos

Que sino Cleon nos va a pegar

AdriAlvarez00 committed:
Ya no te puedes impulsar contra el objeto del que te agarras

Im something of a geometer myself

AdriAlvarez00 committed:
Limpiado un poco

Adrian-de-Lucas-Gomez reviewed (approved):
Que guay. Esto va a solucionar muchas cosas.

WillMDev reviewed (changes requested):
Funciona con los muebles pero pasa lo contrario con los tinky muertos

AdriAlvarez00 committed:
Solucionado bug del cadaver

Al final era una toneria. estabamos cogiendo el angulo de la fixture en vez del del collider.

EstebenR reviewed (changes requested):
Ahora si estas agarrado a un cuerpo solo te deja impulsarte hacia el

AdriAlvarez00 committed:
Merge remote-tracking branch 'origin/master' into 172830483-BugFixMorirAgarrado

AdriAlvarez00 committed:
Merge remote-tracking branch 'origin/172599851-SoltarArmasAlImpactar' into 172830483-BugFixMorirAgarrado

AdriAlvarez00 committed:
Antes el codigo estaba roto. Ahora lo estamos los dos.

AdriAlvarez00 committed:
Todos los enemigos se pueden vencer

Box2D a veces calculaba las normales al reves, pero ahora se comprueban

AdriAlvarez00 committed:
Collider viewer y angulos mas permisivos

Se puede ajustar el angulo con la constante "IMPULSE_ANGLE_ERROR"

AdriAlvarez00 committed:
Fixes del merge

EstebenR reviewed (changes requested):
for some reason la banana esta en el objectFactory

te mete demasiado en las paredes al agarrarte

Adrian-de-Lucas-Gomez committed:
cambiado cuanto te acerca

Evitavos que se incrusten y usamos la mitad del ancho del jugador para acercarlo si es necesario.

EstebenR commented on src/SourceFiles/ObjectFactory.cpp:
```diff
@@ -184,7 +184,7 @@ Entity* ObjectFactory::makeBoiler(EntityManager* entityManager, b2World* physics
 {
 	Entity* e = entityManager->addEntity();
 	Collider* collRoomba = e->addComponent<Collider>(physicsWorld, b2_staticBody, pos.x, pos.y, CONST(double, "BOILER_WIDTH"), CONST(double, "BOILER_HEIGHT"), CONST(double, "BOILER_DENSITY"), CONST(double, "BOILER_FRICTION"),
```
por que el collider del boiler se llama collRoomba? me acabo de dar cuenta xd

Adrian-de-Lucas-Gomez committed:
Un puntero innecesario y método de makeBanana quitado

Estaba en el ObjectFactory y no se usaba porque ya está en el WeaponFactory. El puntero del collider era innecesario.

EstebenR reviewed (approved):
ahora si chavales

WillMDev reviewed (changes requested):
El impulso vuelve a ser spameable.

Adrian-de-Lucas-Gomez committed:
Arreglado abuso del impluso

Ya no puedes fium to fast

AdriAlvarez00 committed:
Merge branch 'master' into 172830483-BugFixMorirAgarrado

WillMDev committed:
Merge branch 'master' into 172830483-BugFixMorirAgarrado
